dbms/src/Storages/DeltaMerge/DeltaMergeStore.cpp

```diff
@@ -750,12 +750,14 @@ void DeltaMergeStore::mergeDeltaAll(const Context & context)
         }
     }
 
+    bool all_succ = true;
     for (auto & segment : all_segments)
     {
-        segmentMergeDelta(*dm_context, segment, MergeDeltaReason::Manual);
+        all_succ = all_succ && segmentMergeDelta(*dm_context, segment, MergeDeltaReason::Manual) != nullptr;
```

Comment:
In this logic, if one segment failed in segmentMergeDelta, then the subsequent segments will not do segmentMergeDelta. It seems not appropriate.